spec/controllers/campaigns_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+require 'rails_helper'
+
+RSpec.describe CampaignsController, type: :controller do
+  let(:campaign) { create(:campaign) }
+
+  describe 'GET #index' do
+    it 'render index template' do
+      get :index
+      is_expected.to render_template(:index)
+    end
+  end
+
+  describe 'GET #show' do
+    it 'render show template' do
+      get :show, params: { id: campaign.id }
+      is_expected.to render_template(:show)
+    end
+  end
+
+  describe 'GET #new' do
+    it 'render new template' do
+      get :new
+      is_expected.to render_template(:new)
+    end
+  end
+
+  describe 'POST #create' do
+    let(:user) { create(:user) }
+    before(:each) do
+      allow(controller).to receive(:current_user).and_return(user)
+    end
+
+    it 'increases number of campaigns' do
+      expect do
+        post :create, params: { campaign: attributes_for(:campaign) }
+      end.to change(Campaign, :count).by(1)
+    end
+
+    it 'redirect to action  index' do
+      post :create, params: { campaign: attributes_for(:campaign) }
+      is_expected.to redirect_to(action: :index)
+    end
+
+    it 'set flash' do
+      post :create, params: { campaign: attributes_for(:campaign) }
+      is_expected.to set_flash
+    end
+  end
+
+  describe 'PUT #update' do
+    let(:title) { Faker::Lorem.characters(3) }
+    it 'redirect to action index' do
+      put :update, params: { id: campaign.id, campaign: attributes_for(:campaign, title: :title) }
+      is_expected.to redirect_to(action: :index)
+    end
+
+    it 'set flash' do
+      put :update, params: { id: campaign.id, campaign: attributes_for(:campaign, title: :title) }
+      is_expected.to set_flash
+    end
+  end
+
+  describe 'DELETE #destroy' do
+    it 'do not change number of campaigns' do
+      expect do
+        delete :destroy, params: { id: campaign.id }
+      end.to_not change(Campaign, :count)
```

Comment:
why do you say `to_not change(Campaign, :count)`? it should change a number of campaigns since it deletes it from the database.